include/range/v3/utility/iterator.hpp

```diff
@@ -389,8 +423,9 @@ namespace ranges
             RANGES_CXX14_CONSTEXPR
             iterator_size_t<I> operator()(I begin, S end) const
             {
-                RANGES_ASSERT(0 <= (end - begin));
-                return static_cast<iterator_size_t<I>>(end - begin);
+                auto n = static_cast<iterator_difference_t<I>>(end - begin);
```

Comment:
Doesn't the expression `end - begin` already have type `iterator_difference_t<I>`?